src/coverlet.core/Reporters/CoberturaReporter.cs

```diff
@@ -133,28 +133,48 @@ public string Report(CoverageResult result)
             return Encoding.UTF8.GetString(stream.ToArray());
         }
 
-        private static IEnumerable<string> GetRootDirs(Modules modules, bool useSourceLink)
+        private static IEnumerable<string> GetBasePaths(Modules modules, bool useSourceLink)
         {
             if (useSourceLink)
             {
                 return new[] { string.Empty };
             }
 
-            return modules.Values.SelectMany(k => k.Keys).Select(Directory.GetDirectoryRoot).Distinct();
+            var pathsGroupedByRootDir = modules.Values.SelectMany(k => k.Keys).GroupBy(Directory.GetDirectoryRoot);
```

Comment:
`pathsGroupedByRootDir` is used only in one place use it directly
```cs
           return modules.Values.SelectMany(k => k.Keys).GroupBy(Directory.GetDirectoryRoot).Select(group =>
            {
               ...
```

Reply:
Done